packages/block-library/src/navigation/edit/use-create-navigation-menu.js

```diff
@@ -68,6 +68,16 @@ export default function useCreateNavigationMenu( clientId ) {
 				.then( ( response ) => {
 					setValue( response );
 					setStatus( CREATE_NAVIGATION_MENU_SUCCESS );
+
+					// Set the status to publish so that the Navigation block
+					// shows up in the multi entity save flow.
+					editEntityRecord(
+						'postType',
+						'wp_navigation',
+						response.id,
+						{ status: 'publish' }
+					);
```

Comment:
I think it'd be good to do something like this:
```suggestion
                    if ( postStatus !== 'publish' ) {
					    editEntityRecord(
						    'postType',
						    'wp_navigation',
						    response.id,
						    { status: 'publish' }
					    );
					}
```

I see that when converting a classic menu, the status is set straight to `publish` so this step could be skipped.